src/app/Plans/components/PlansTable.tsx

```diff
@@ -1,19 +1,119 @@
+import { useSortState, usePaginationState } from '@app/common/hooks';
+import { IPlan, IMigration } from '@app/queries/types';
 
 interface IPlansTableProps {
-  planList: IPlan[];
-  toggleAddWizardOpen: () => void;
+  plans: IPlan[];
+  migrations: IMigration[];
 }
 
 const PlansTable: React.FunctionComponent<IPlansTableProps> = ({
-  planList,
-  toggleAddWizardOpen,
+  plans,
+  migrations,
 }: IPlansTableProps) => {
+  const getSortValues = (plan: IPlan) => {
+    return [
+      plan.metadata.name,
+      plan.spec.provider.sourceProvider.name,
+      plan.spec.provider.destinationProvider.name,
+      plan.spec.vmList.length,
+      '', // Plan status
+      '', // Action column
+    ];
+  };
+
+  const { sortBy, onSort, sortedItems } = useSortState(plans, getSortValues);
+  const { currentPageItems, setPageNumber, paginationProps } = usePaginationState(sortedItems, 10);
+  React.useEffect(() => setPageNumber(1), [sortBy, setPageNumber]);
+
+  const getMigration = (plan) => migrations.filter((migration) => migration.plan === plan);
+
+  const columns: ICell[] = [
+    { title: 'Name', transforms: [sortable, wrappable] },
+    { title: 'Source provider', transforms: [sortable, wrappable] },
+    { title: 'Target provider', transforms: [sortable] },
+    { title: 'VMs', transforms: [sortable] },
+    { title: 'Plan status', transforms: [sortable] },
+    { title: '' },
+  ];
+
+  const rows: IRow[] = [];
+
+  currentPageItems.forEach((plan: IPlan) => {
+    let buttonText = '';
+    if (plan.status.conditions.every((condition) => condition.type === 'Ready')) {
+      buttonText = 'Start migration';
+    }
+
+    rows.push({
+      meta: { plan },
+      cells: [
+        {
+          title: (
+            <>
+              <a href="#">{plan.metadata.name}</a>
+              <div>{plan.spec.description}</div>
+            </>
+          ),
+        },
+        plan.spec.provider.sourceProvider.name,
+        plan.spec.provider.destinationProvider.name,
+        plan.spec.vmList.length,
+        {
+          title: <PlanStatus plan={plan} migration={getMigration(plan)[0]} />,
+        },
+        {
+          title: buttonText ? (
+            <>
+              <Button variant="secondary" onClick={() => alert('TODO')} isDisabled={false}>
+                {buttonText}
+              </Button>
+              <PlanActionsDropdown />
+            </>
+          ) : null,
+        },
+      ],
+    });
+  });
+
   return (
-    <div className={`test-component test-component-secondary`}>
-      <h1 className="heading">Migration Plan table</h1>
-    </div>
+    <>
+      <Level>
+        <LevelItem>
+          <Button variant="secondary" onClick={() => alert('TODO')} isDisabled={false}>
+            Create
+          </Button>
+        </LevelItem>
+        <LevelItem>
+          <Pagination {...paginationProps} widgetId="plans-table-pagination-top" />
+        </LevelItem>
+      </Level>
+      <Table
```

Comment:
Let's also add a CSS file for the PlansTable so we can apply some custom alignment. If you look at the mockups, the cell contents are vertically centered, and in this code the columns in the middle are aligned with the top of the cell, which looks a little odd.

I think if we put a `className="plans-table"` on here and then add this CSS, it will fix it:

```css
.pf-c-table.plans-table tbody > tr > * {
  vertical-align: center
}
```

This will override the `vertical-align: baseline` style that PF provides here (a weird choice IMO). There might be a more React-y way to do this, not sure.